src/components/figures/chart/chartJs/chartjs.ts

```diff
@@ -49,8 +49,13 @@ export class ChartJsComponent extends Component<Props, SpreadsheetChildEnv> {
     useEffect(() => {
       const runtime = this.chartRuntime;
       if (!deepEquals(runtime, this.currentRuntime, "ignoreFunctions")) {
+        if (runtime.chartJsConfig.type !== this.currentRuntime.chartJsConfig.type) {
+          this.chart?.destroy();
+          this.createChart(deepCopy(runtime.chartJsConfig));
+        } else {
+          this.updateChartJs(deepCopy(runtime));
```

Comment:
Because before the ChartJS runtime type changed only when the definition's type changed. And when the definition's type changes, we create a new component. 

Here we stay in the same component, so we need to re-create a chart